### PR TITLE
Implement forgot your password interface for doctrine.

### PR DESCRIPTION
**Pull request by lebacql**

### CODE REVIEW

File: src/UserDao.php [hunk 1]
```diff
@@ -43,29 +47,31 @@ public function getUserByCredentials($login, $password) {
...
+	 * @throws UserDaoException
+	 */
 	public function getUserByToken($token) {
-		$user = $this->findByToken(
-				$token
-		);
+		$user = $this->findBy([
+				'token' => $token
+		]);
+
```
Comment (xhuberty) [inline src/UserDao.php]: Use a findUniqueByToken($token); And clean the useless code below.

File: src/UserDao.php [hunk 2]
```diff
@@ -92,13 +98,13 @@ public function getUserById($id) {
...
 	 * Returns a user from its login
-	 *
 	 * @param string $login
-	 * @return UserInterface
-	*/
+	 * @return null
+	 * @throws UserDaoException
+	 */
 	public function getUserByLogin($login) {
 		$user = $this->findByLogin(
```
Comment (xhuberty) [inline src/UserDao.php:103]: Same as below.

File: src/UserDao.php [hunk 3]
```diff
@@ -110,5 +116,46 @@ public function getUserByLogin($login) {
...
+	 * Sets $token for user whose mail is $email, stores the token in database.
+	 * Throws an EmailNotFoundException if the email is not part of the database.
+	 *
+	 * @param string $email
+	 *
+	 * @throws \Mouf\Security\Password\Api\EmailNotFoundException
+	 */
+	public function setToken(string $email, string $token)
+	{
+		$user = $this->findOneBy([
```
Comment (xhuberty) [inline src/UserDao.php:124]: Use findUniqueBy

**Comment by Ngob** [comment]
Uncorrect target branch

**Pull request closed without merging**